Test/Tests/Network/IPFabric.Tests.ps1

```diff
@@ -25,23 +25,60 @@ Param (
 . $PSScriptRoot\..\..\Utils\DockerNetwork\Commands.ps1
 
 $ContrailProject = 'ci_tests_ip_fabric'
-$DockerImage = 'microsoft/windowsservercore'
 $ContainerID = 'jolly-lumberjack'
 $ContainerNetInfo = $null
 
-$Subnet = [Subnet]::new(
-    '172.16.0.128',
-    28,
-    '172.16.0.129',
-    '172.16.0.130',
-    '172.16.0.140'
-)
-
 $NetworkPolicy = [NetworkPolicy]::new_PassAll('passallpolicy', $ContrailProject)
-$VirtualNetwork = [VirtualNetwork]::New('testnet_fabric_ip', $ContrailProject, $Subnet)
+$VirtualNetwork = [VirtualNetwork]::New('testnet_fabric_ip', $ContrailProject, $null)
 $VirtualNetwork.NetworkPolicysFqNames = @($NetworkPolicy.GetFqName())
 $VirtualNetwork.EnableIpFabricForwarding()
 
+function Get-ContainerSubnet {
+    Param(
+        [Parameter(Mandatory = $true)] [Hashtable] $IpInfo
+    )
+
+    function Convert-IpUInt32ToString {
+        Param([Parameter(Mandatory = $true)] [uInt32] $Ip)
+
+        [String[]] $Out = [String[]]::new(4)
+        foreach ($i in (3..0)) {
+            $Out[$i] = [convert]::ToString($Ip -band 255)
+            $Ip = $ip -shr 8
+        }
+        return ($out -join '.')
+    }
+
+    function Convert-IpStringToUInt32 {
+        Param([Parameter(Mandatory = $true)] [String] $Ip)
+
+        [String[]] $IPAddressArr = $Ip.Split('.')
+        [uInt32] $Out = 0
+        foreach ($i in (3..0)) {
+            $Out += ([convert]::ToUInt32($IPAddressArr[3 - $i]) -shl $i * 8)
+        }
+        return $Out
+    }
+
+    $NetPrefixMask = [uInt32]::MaxValue -shl (32 - $IpInfo.PrefixLength)
+    $NetIp = Convert-IpStringToUInt32 -Ip $IpInfo.IPAddress
+    $NetPrefix = $NetIp -band $NetPrefixMask
+    $NetBroadcast = $NetPrefix + (-bnot $NetPrefixMask)
+
+    $SubnetPoolEnd = $NetBroadcast - 1
+    $SubnetPoolBeg = $NetBroadcast - 1
+    $SubnetDefaultGate = $NetBroadcast - 6
+    $SubnetPrefix = $NetBroadcast - 7
```

Comment:
please add some comments here, especially why the new PrefixLength is 29

Reply:
Done